code/src/main/java/com/onlinecontacttracing/messaging/EmailSender.java

```diff
@@ -0,0 +1,100 @@
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  private static final List<String> SCOPES = Collections.singletonList(GmailScopes.GMAIL_SEND);
+  //rename to be like authconfigurationdata, also need to remove from github
+  private static final String CREDENTIALS_FILE_PATH = "/credentials.json";
+  // final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+  
+  public EmailSender(String emailSubject, ArrayList<PotentialContact> contactsList) {
+    this.emailSubject= emailSubject;
+    this.contactsList = contactsList;
+    // service = new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, getCredentials(HTTP_TRANSPORT, user))
+    //             .setApplicationName(APPLICATION_NAME)
+    //             .build();
+    this.user = user;
+  }
+
+  public void setMessageObject(SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage, PositiveUser user) {
+    this.messageObject = new MessageServlet(systemMessage, localityResource, customizableMessage, user);
+  }
+
+  public void setEmailBody(String messageLanguage) {
+    this.messageObject.compileMessages(messageLanguage);
+    this.emailBody = this.messageObject.getCompiledBackendMessage();
+  }
+
+  public void sendEmailsOut() {
+    PotentialContact contact;
+    MimeMessage email;
+    for(PotentialContact contactName : this.contactsList) {
+      // email = MessagingSetup.createEmail(contactName.getEmail(), user.getUserEmail(), emailSubject, emailBody);
+      // sendMessage(service, user.getUserId(), email);
+    }
+  }
+
+  private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT, PositiveUser user) throws IOException {
+        // Load client secrets
+        //create a class authAPI config constants, have this as a string, eliminate input reader
+        InputStream in = EmailSender.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
+        if (in == null) {
+            throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+        }
+        GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+      
+        TokenResponse tokenResponse = new TokenResponse().setScope(SCOPES.get(0));
+        return new GoogleAuthorizationCodeFlow.Builder(
+               HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES).build().createAndStoreCredential(tokenResponse, user.getUserId());
+        //  = new GoogleAuthorizationCodeFlow.Builder(
+        //         HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+        //         .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+        //         .setAccessType("offline")
+        //         .build();
+        // LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8888).build();
+        // return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+    }
```

Comment:
If I understand correctly, this is still getting the credentials for the sending user, rather than for our service account. Is that right?